pkg/operator/operator.go

```diff
@@ -133,6 +137,34 @@ func New(ctx context.Context, config operatorconfig.Config, kubeConfig *rest.Con
 		return nil, fmt.Errorf("failed to create status controller: %v", err)
 	}
 
+	// Set up the DNSNameResolver controller. This controller is unmanaged by
+	// the manager; the enable_dnsnameresolver controller starts it and the
+	// caches.
```

Comment:
Please mention in the comments: 
- the two caches represented in `dnsNameResolverCaches`: `dnsNameResolverCache` and `corednsEndpointsSliceCache`.
- some information about what this controller is for, e.g. relevant details from https://github.com/openshift/coredns-ocp-dnsnameresolver/blob/5fa57d974c2285163d246ca38fd1e9f96922df31/operator/controller/dnsnameresolver/controller.go#L49-L53